### DISCUSSION TITLE
help with simple disease simulation

### DISCUSSION THREAD

**Discussion by fusougod** [gama-platform/gama#167] [Q&A]
hey! im pretty much new into using GAMA, i was wondering if i could get some help with my code. 

model prototypedisease 

    global {
        int initial_infected <- 10; 
        int population_size <- 1000; 
        float infection_rate <- 0.03; 
        float recovery_rate <- 0.05; 
    }

    species person {
        float x <- rnd(100); 
        float y <- rnd(100); 
        string health_status <- "susceptible"; 

        if health_status = "susceptible" {
            if any(agent.neighbors() where _.health_status = "infected") {
                if rnd(1) < infection_rate {
                    set health_status <- "infected";
                }
            }
        } else if health_status = "infected" {
            if rnd(1) < recovery_rate {
                set health_status <- "recovered";
            }
        }

        draw {
            color <- if health_status <- "susceptible" then "green" else if health_status = "infected" then "red" else "blue";
            radius <- 2;
        }

    
        space {
            grid {
                x <- 100;
                y <- 100;
            }
        }

       
        schedule {
            daily {
                move;
            }
        }
    }

    create n_initial_infected person agents {
        start {
            set health_status <- "infected";
        }
    }

    create (population_size - n_initial_infected) person agents {
        start {
            set health_status <- "susceptible";
        }
    }
}

i seem to get an error on these lines, may i ask what is the issue, and how do i fix it?
![image](https://github.com/gama-platform/gama/assets/98147172/1e7acf22-1594-42b5-afa0-e021ce6fc939)
![image](https://github.com/gama-platform/gama/assets/98147172/3acd77d2-6787-447d-8fdd-56669f697997)

**Comment by lesquoyb** [comment]
Hello,
I guess you generated that with an ai or something, this is just not the gaml syntax, that's why you have error.
If you are interested in writing models in gaml you can have a look at the documentation it's very intuitive, here is how to write an [if statement](https://gama-platform.org/wiki/next/BasicProgrammingConceptsInGAML#conditional-structures), else sorry but ai are not good at writing gaml code currently so you'll have to find another way
